lms/djangoapps/lms_xblock/services.py

```diff
@@ -0,0 +1,73 @@
+"""
+Module implementing XBlock runtime services.
+
+TODO: services should be provided as entry points, rather than hard-coded.
+"""
+
+from openedx.core.djangoapps.user_api.course_tag import api as user_course_tag_api
+from xmodule.modulestore.django import modulestore
+from xmodule.partitions.partitions_service import PartitionService
+
+
+class LmsPartitionService(PartitionService):
+    """
+    Another runtime mixin that provides access to the student partitions defined on the
+    course.
+
+    (If and when XBlock directly provides access from one block (e.g. a split_test_module)
+    to another (e.g. a course_module), this won't be necessary, but for now it seems like
+    the least messy way to hook things through)
+
+    """
+    @property
+    def course_partitions(self):
+        course = modulestore().get_course(self._course_id)
+        return course.user_partitions
+
+
+class UserTagsService(object):
+    """
+    A runtime class that provides an interface to the user service.  It handles filling in
```

Comment:
You mean "an interface to the user **tags** service" right? The user service is `DjangoXBlockUserService`/`UserService` which seems like a different thing.